src/hooks/useDarkMagic.ts

```diff
@@ -37,7 +39,23 @@ const pipx = async (arg0: string | undefined) => {
     return {
       project: "pypa.github.io/pipx",
       constraint: new Range("*"),
-      shebang: ["pipx", "run"]
+      shebang: ["pipx", "run"],
+      precmd: undefined
+    }
+  }
+
+  return undefined
+}
+
+const cargo = async (arg0: string | undefined) => {
+  const res = await useFetch(`https://crates.io/api/v1/crates/${arg0}`)
+
+  if (res.status == 200) {
+    return {
+      project: "rust-lang.org/cargo",
+      constraint: new Range("*"),
+      shebang: undefined,
+      precmd: ["cargo", "install", arg0!]
```

Comment:
installs to `~/.local/bin/foo`, presumably we should explicitly execute that?

Reply:
We've already shown it's not on the path before installing, so that's safer but shouldn't be at issue. *Shouldn't be*...